doc/1.7/dev_guide/lua_style_guide.rst

```diff
@@ -0,0 +1,465 @@
+===========================================================
+
+* 4 spaces instead tabs. PIL suggests using of two spaces, but programmer looks
+  at code 4 up to 8 hours a day, so it's simplier to distinguish indentation
+  with 4 spaces. Why spaces? Similar representation everywhere.
+
+  You can use vim modelines:
+
+  .. code-block:: lua
+
+      -- vim:ts=4 ss=4 sw=4 expandtab
+
+* No whitespaces at EOL, newline at end of file (?)
+
+* Every do/while/for/if/function should indent 4 spaces.
+
+* related ``or``/``and`` in ``if`` must be enclosed in the round brackets (). Example:
+
+  .. code-block:: lua
+
+      if (a == true and b == false) or (a == false and b == true) then
+          <...>
+      end -- good
+
+      if a == true and b == false or a == false and b == true then
+          <...>
+      end -- bad
+
+      if a ^ b == true then
+      end -- good, but not explicit
+
+* Type conversion
+
+  Do not use concatenation to convert to string or addition to convert to number
+  (use ``tostring``/``tonumber`` instead):
+
+  .. code-block:: lua
+
+      local a = 123
+      a = a .. ''
+      -- bad
+
+      local a = 123
+      a = tostring(a)
+      -- good
+
+      local a = '123'
+      a = a + 5 -- 128
+      -- bad
+
+      local a = '123'
+      a = tonumber(a) + 5 -- 128
+      -- good
+
+* Try to avoid multiple nested if's or conditions with common body:
+
+  .. code-block:: lua
+
+      if (a == true and b == false) or (a == false and b == true) then
+          do_something()
+      end
+      -- good
+
+      if a == true then
+          if b == false then
+              do_something()
+          end
+      if b == true then
+          if a == false then
+              do_something()
+          end
+      end
+      -- bad
+
+* Avoid multiple concatenations in one parameter, use string.format instead:
+
+  .. code-block:: lua
+
+      function say_greeting(period, name)
+          local a = "good  " .. period .. ", " .. name
+      end
+      -- bad
+
+      function say_greeting(period, name)
+          local a = string.format("good %s, %s", period, name)
+      end
+      -- good
+
+      local say_greeting_fmt = "good %s, %s"
+      function say_greeting(period, name)
+          local a = say_greeting_fmt:format(period, name)
+      end
+      -- best
+
+* ``and``/``or`` for default variable values or..
+
+  .. code-block:: lua
+
+    function(input)
+        input = input or 'default_value'
+    end -- good
+
+    function(input)
+        if input == nil then
+            input = 'default_value'
+        end
+    end -- ok, but excessary
+
+* If's and simple cases:
+
+  .. code-block:: lua
+
+      if a == true then
+          return do_something()
+      end
+      do_other_thing() -- good
+
+      if a == true then
+          return do_something()
+      else
+          do_other_thing()
+      end -- bad
+
+* Using spaces:
+
+  - do not use a space after functions and after commas in arguments:
```

Comment:
I didn’t understand what stated here. Do not use a space btw a function name and opening circle bracket? Use a space after commas in formal arguments list?